tests/accelerators/test_ipu.py

```diff
@@ -200,20 +200,13 @@ class TestCallback(Callback):
 
         def on_train_start(self, trainer: Trainer, pl_module: LightningModule) -> None:
             assert trainer.accelerator.model.precision == 16
-            assert trainer.accelerator.training_type_plugin.convert_model_to_half
             for param in trainer.accelerator.model.parameters():
                 assert param.dtype == torch.float16
             raise SystemExit
 
     model = IPUModel()
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        fast_dev_run=True,
-        ipus=1,
-        precision=16,
-        plugins=IPUPlugin(convert_model_to_half=True),
-        callbacks=TestCallback()
-    )
+    model = model.half()
```

Comment:
Curious. Why not half it as a flag?

Reply:
its a bit boilerplate-ish and not very pytorch-esque. It's a bit confusing why you can't just convert your model to half which is the pytorch way and what we really do internally :)